packages/ui-kit/src/components/multi-code-block.tsx

```diff
@@ -193,23 +193,25 @@ function MultiCodeBlock(props: MultiCodeBlockProps) {
         })}
     </Container>
   );
-}
-
-export default MultiCodeBlock;
+};
 
 const getCodeBlockPropsFromMdxPreNodeProps = (props: {
   children?: React.ReactNode;
 }) => {
-  const childElement = reactIs.isElement(props.children)
-    ? props.children
-    : null;
+  const childElement =
+    props.children && reactIs.isElement(props.children) ? props.children : null;
   const childProps = childElement?.props;
   const className = childProps ? childProps.className : '';
   const languageToken: string = className || 'language-text';
   const [, languageCode] = languageToken.split('language-');
   const parsedOptions = parseCodeBlockOptions(childProps);
+
   const content =
-    childProps && childProps.children ? childProps.children : childProps;
+    childProps && childProps.children
+      ? Array.isArray(childProps.children)
+        ? childProps.children[0]
+        : childProps.children
+      : childProps;
```

Comment:
I didn't go through the whole thing, but is there a better way than the fixed index here?